src/v/v8_engine/script_dispatcher.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "model/metadata.h"
+#include "utils/mutex.h"
+#include "v8_engine/script.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/util/variant_utils.hh>
+
+#include <absl/container/node_hash_map.h>
+#include <absl/hash/hash.h>
+
+#include <exception>
+#include <iostream>
+#include <variant>
+
+namespace v8_engine {
+
+struct script_compile_params {
+    ss::sstring function_name;
+    iobuf code;
+    size_t max_heap_size_bytes{10000};
+    size_t run_timeout_ms{200};
+};
+
+// This class implement database for v8_runtimes. It can be updated by
+// data_policy_manager. Fetch request will only run get for getting v8_runtimes
+template<typename Executor>
+class script_dispatcher {
+    constexpr static size_t _mutex_count{50};
```

Comment:
comment explaining what this is, also why is it not configurable, magic number etc